### ISSUE TITLE
Api.reauthenticate does not work

### ISSUE THREAD

**Issue by yzhaoa** [kr8s-org/kr8s#306]
### Which project are you reporting a bug for?

kr8s

## Background

Due to https://github.com/kr8s-org/kr8s/issues/125 not being implemented yet, when kr8s is being used with expiring auth tokens, e.g. those issued by gke-gcloud-auth-plugin, will stop working after token expiration.

Reading the source code, it seems that the Api object has a per-thread cache, making it so that once the token expires, the Api becomes completely unusable for a thread.

This is perhaps not an issue for short-lived command-line tools, but for using kr8s e.g. in jupyter notebooks, it is very annoying.

Above is expected and I'll leave it to https://github.com/kr8s-org/kr8s/issues/125 for a proper fix.

## Expected

To work around the above issue, I tried to call `Api.reauthenticate` whose name and docstring indicate that it should be able to solve the problem and refresh the auth token and make kr8s usable again from a thread.

## Actual

Api.reauthenticate doesn't do anything. `Api.auth.token` does not get refreshed after the call.

## Bug

https://github.com/kr8s-org/kr8s/blob/a602e53e2b5bc753bc2989fbc9dd60ada6add561/kr8s/_auth.py#L55:

```python
            if self._kubeconfig is not False and not self.server:
                await self._load_kubeconfig()
```

Because _load_kubeconfig sets self.server, the if clause never passes, and _load_kubeconfig() is never called again.

## Workaround

```python

api = kr8s.api(...)

for attempt in range(2):
  if attempt == 1:
    # Remove after https://github.com/kr8s-org/kr8s/issues/125  is fixed.
    api.auth.server = None  # Hack
    api.reauthenticate()


  try:
    # do something with the api
    pass
  except kr8s.ServerError as exc:
    if exc.status['code'] != 401:
      raise
  else:
    break
else:
  raise ValueError(f'Failed to resolve auth failure')

```

**Comment by jacobtomlinson** [comment]
I don't actually think this is related to #125 as that covers refreshing an OIDC token which is a long winded process that we haven't implemented yet. 

The problem here is that the exec command will refresh the token, but we are not calling it again.

The problem seems to be because we are using `self.server` to store both the url that a user can pass in, as well as the url detected from the config. We check if the url has been set so that we don't overwrite what the user may have passed in, but after the first auth we can't tell the difference between manually setting the url and programatically reading it from the kube config.

I think I can open a quick PR to resolve this.

**Comment by jacobtomlinson** [comment]
I just released `v0.13.2` which contains the fix for this. You shouldn't need to worry about calling `reauthenticate()` at all now, when your token expires and a request fails with a `401` or `403` it will try and refresh the tokens automatically and then retry the request. So it should reauthenticate transparently.
